### DISCUSSION TITLE
Can we have a new field of phone number input field that displays the flags of the countries

### DISCUSSION THREAD

**Discussion by nouaim** [primefaces/community#2764] [PrimeNG]
I propose Primeng builds a new field for phone number input which displays the flags of the countries and lets the user chooses the country that corresponds to its phone number. 

Example: 
![Screenshot from 2024-09-28 10-37-33](https://github.com/user-attachments/assets/bbb5388a-5406-4d0a-b716-aaf907133e8a)

A lot of open source projects have built it but most of them didn't upgrade to be used with the last version of Angular, and I found one of them that can be used with Angular 18 but its theme is based on material and doesn't fit well with the theme of Primeng, and I leave its url here: 

https://github.com/juanjotorres90/ngx-material-intl-tel-input
its demo: https://juanjotorres.net/projects/ngx-material-intl-tel-input/ 

This feature is really nice to have in primeng as well especially that it has phone validation and displays an alert message when the  user enters a wrong phone number (it uses phone validation from an open source library that Google has built)

What do you think of this proposition? and if it is useful how can we start collaborating in building this feature?

**Comment by nazem0** [comment]
@nouaim 
Implemented externally by [rafael-faitao](https://github.com/rafael-faitao)
https://github.com/rafael-faitao/ngx-p-world-phone-input

**Reply by nouaim** [reply]
@nazem0 you can notice version compatibility is with Angular 16 only in the project you shared, and Angular will upgrade to version 21 in November, do you see the gap? 

If you read the text above, I've included an open source project which can be used with Angular's new versions, however its theme is not compatible with primeng, and it uses Angular Material. So if you use Tailwind alone for example as your style library, you will end up importing another css library (material) for the purpose of one field only (phone number field), not to mention the theme incompatibility that will enforce you to override a lot of the field's css classes to have something close to primeng theme (in short, the work won't be clean), that's why it is good to implement one for primeng.
